near-sdk/src/environment/env.rs

```diff
@@ -553,15 +554,48 @@ pub fn promise_batch_action_add_key_with_full_access(
         )
     }
 }
+
+/// This is a short lived function while we migrate between the Balance and the allowance type
+pub(crate) fn migrate_to_allowance(allowance: Balance) -> Allowance {
+    match allowance {
+        0 => Allowance::Unlimited,
+        x => Allowance::limited(x).expect("This must be non zero"),
+    }
```

Comment:
feels like this should be:

```suggestion
    Allowance::limited(allowance).unwrap_or(Allowance::Unlimited)
```

Sorry if that doesn't compile, I didn't check it, but want to avoid the expect which comes with unnecessary bloat to compiled code size